src/app/application.cpp

```diff
@@ -196,6 +196,9 @@ Application::Application(int &argc, char **argv)
 
     if (m_commandLineArgs.webUiPort > 0) // it will be -1 when user did not set any value
         Preferences::instance()->setWebUiPort(m_commandLineArgs.webUiPort);
+
+    if (m_commandLineArgs.torrentingPort > 0) // it will be -1 when user did not set any value
+        Preferences::instance()->setTorrentingPort(m_commandLineArgs.torrentingPort);
```

Comment:
You should use `SettingValue` to update preference.
And revert changes in `Preference` class.

```suggestion
        SettingValue<int>(u"BitTorrent/Session/Port"_qs) = m_commandLineArgs.torrentingPort;
```

Reply:
In case the code above didn't work:
```c++
if (m_commandLineArgs.torrentingPort > 0) // it will be -1 when user did not set any value
{
    SettingValue<int> port {u"BitTorrent/Session/Port"_qs};
    port = m_commandLineArgs.torrentingPort;
}
```

Reply:
Dear Chocobo1! I see, thank you for checking.
Guess I caused more trouble with this few liner than help. Kind of expected of me.

I based it on the Web UI Port completely, while paying attention not to break things.

But uh, can I ask why the difference between the two?
Like, if the WebUiPort does the call like that... is that wrong over there? Or, the WebUi settings are handled in a different way?

I don't mean to argue I just wish to understand, learn more about the code. Don't want to waste your time (besides making you read this) so a short tl;dr is totally fine too, I'm just curious. Thank you!

Reply:
>But uh, can I ask why the difference between the two?

It is actually a train of thoughts:
The torrenting port is (kind of) managed by the Session class. It would be expected to use `Session:setPort()` to update values and not introduce setter/getter in `Preference` class.
However, using `Session:setPort()` would produce an unwanted effect: Since the method can only be called after initializing an instance of `Session`. `Session` constructor will first use the old port and then quickly switch over to the new port and this is undesirable. So the only way (*without refactoring the code base* which I have no idea how) is to overwrite the setting directly. To do this it is preferred to use `SettingValue`.

>Guess I caused more trouble with this few liner than help. Kind of expected of me.

Don't mind it, discussions and suggestions is expected when conducting code review. It happens to everyone.
And it is reasonable to ask for clarifications, just ask away.

Reply:
> So the only way (_without refactoring the code base_ which I have no idea how) is to overwrite the setting directly.

Sorry, I didn't follow this topic in detail. But I have a general question. Is overriding some settings via the command line supposed to be applied permanently? I would expect only a temporary effect for the current run.
But it looks like it requires additional logic to correctly handle it. It seems to be easier using separate Preferences component.

Reply:
> `Session` constructor will first use the old port and then quickly switch over to the new port and this is undesirable.

You could add optional `port` parameter to Session constructor in order to redefine it.

Reply:
>But I have a general question. Is overriding some settings via the command line supposed to be applied permanently? I would expect only a temporary effect for the current run.

I think it depends. IIRC users complained about qbt didn't show correctly the currently in-use port in the Options. So it would be better that we just change the option than making it temporary (or more complicated).